lib/datadog/core/configuration/option.rb

```diff
@@ -14,7 +14,7 @@ class Option
         # @!attribute [r] precedence_set
         #   When this option was last set, what was the value precedence used?
         #   @return [Precedence::Value]
-        attr_reader :definition, :precedence_set
+        attr_reader :definition, :precedence_set, :effective_env
```

Comment:
Nit, non-blocking: not sure the name is intuitive? `effective` can be ambiguous as to whether its present or past tense. I think I'd like to see something that demonstrates the "ex-post-facto" nature of this value. Top of mind suggestion: `resolved_env`? `env_used`?